src/test/java/com/exasol/adapter/document/files/AbstractDocumentFilesAdapterIT.java

```diff
@@ -44,30 +45,41 @@
 
     protected abstract void uploadDataFile(final Supplier<InputStream> resource, String resourceName);
 
-    private void createVirtualSchema(final String schemaName, final String resourceName) {
-        createVirtualSchema(schemaName, () -> AbstractDocumentFilesAdapterIT.class.getClassLoader()
-                .getResourceAsStream(IT_RESOURCES + resourceName));
+    private void createVirtualSchemaWithMappingFromResource(final String schemaName, final String resourceName,
+            final String dataFilesDir) throws IOException {
+        final String mappingTemplate = getMappingTemplate(resourceName);
+        final String filledMapping = mappingTemplate.replace("DATA_FILES_DIR", dataFilesDir);
+        createVirtualSchema(schemaName, filledMapping);
     }
 
-    protected abstract void createVirtualSchema(String schemaName, Supplier<InputStream> mapping);
+    private String getMappingTemplate(final String resourceName) throws IOException {
+        try (final InputStream stream = AbstractDocumentFilesAdapterIT.class.getClassLoader()
+                .getResourceAsStream(IT_RESOURCES + resourceName)) {
+            return new String(Objects.requireNonNull(stream).readAllBytes(), StandardCharsets.UTF_8);
+        }
+    }
+
+    protected abstract void createVirtualSchema(String schemaName, String mapping);
 
     @Test
-    void testReadJson() throws SQLException {
-        createJsonVirtualSchema();
+    void testReadJson() throws SQLException, IOException {
+        final String dataFilesDirectory = String.valueOf(System.currentTimeMillis());
```

Comment:
Maybe it is good idea to put this into beforeEach method. If it is directory also create it in that method using TempDir.